docs/topics/OAuth2.md

```diff
@@ -56,68 +43,232 @@ Scopes provide access to certain resources of a user's account. Your API client
+```
 
-Bots within the Discord API are a form of user account that is authenticated _without_ a username or password, and has similar properties and abilities to normal user accounts. Bot accounts enable developers to have a simple portal that allows authenticated users to add third-party bots to servers they own or manage.
+Boom; fresh [access token response](#DOCS_OAUTH2/authorization-code-grant-access-token-response)!
 
-### Registering Bots
+## Implicit Grant
 
-Bots can be registered by clicking the "add bot" button when editing or creating an [OAuth2 application](#MY_APPLICATIONS/top).
+The implicit OAuth2 grant is a simplified flow optimized for in-browser clients. Instead of issuing the client an authorization code to be exchanged for an access token, the client is directly issued an access token. The URL is formatted as follows:
 
-### Two-Factor Authentication Requirement
+###### Implicit URL Example
 
-For bots with [elevated permissions](#DOCS_PERMISSIONS/bitwise-permission-flags) (permissions with a * next to them), we enforce two-factor authentication for the owner's account when used on guilds that have server-wide 2FA enabled.
+```
+https://discordapp.com/oauth2/authorize?response_type=token&client_id=A&scope=B
+```
 
-### Adding Bots to Guilds
+On redirect, your redirect URI will contain additional **URI fragments**: `access_token`, `token_type`, `expires_in`, and `scope`. **These are not querystring parameters.** Be mindful of the "#" character:
 
-A URL can be generated that redirects authenticated users to the add-bot flow. It can be in the following format without a `redirect_uri`:
+###### Implicit Redirect URL Example
 
 ```
-https://discordapp.com/api/oauth2/authorize?client_id=157730590492196864&scope=bot&permissions=0
+https://findingfakeurlsisprettyhard.tv/#access_token=A&token_type=Bearer&expires_in=B&scope=C
 ```
 
-Or with a `redirect_uri`:
+There are tradeoffs in using the implicit grant flow. It is both quicker and easier to implement, but rather than exchanging a code and getting a token returned in a secure HTTP body, the access token is returned in the URI fragment, which makes it possibly exposed to unauthorized parties. If you choose to implement the implicit grant, we highly recommend using the `state` parameter to help protect against cross-site request forgery. You also are not returned a refresh token, so the user must explicitly re-authorize once their token expires.
+
+### Client Credentials Grant
+
+The client credential flow is a quick and easy way for bot developers to get their own bearer tokens for testing purposes. By making a `POST` request to the [token URL](#DOCS_OAUTH2/shared-resources-oauth2-urls) with a grant type of `client_credentials`, you will be returned an access token for the bot owner. Therefore, always be super-extra-very-we-are-not-kidding-like-really-be-secure-make-sure-your-info-is-not-in-your-source-code careful with your `client_id` and `client_secret`.
```

Comment:
last line is a general recommendation for client secrets, and is not strictly for client creds grant